src/libraries/System.Private.CoreLib/src/System/Reflection/FunctionPointerInfo.cs

```diff
@@ -0,0 +1,94 @@
+
+namespace System.Reflection
+{
+    internal sealed partial class FunctionPointerInfo
+    {
+        private const string CallingConventionTypePrefix = "System.Runtime.CompilerServices.CallConv";
+        private readonly RuntimeFunctionPointerParameterInfo _returnInfo;
+        private readonly RuntimeFunctionPointerParameterInfo[] _parameterInfos;
+        private Type[]? _callingConventions;
+
+        public FunctionPointerParameterInfo ReturnParameter => _returnInfo;
+        public FunctionPointerParameterInfo[] ParameterInfos => CloneArray(_parameterInfos);
+
+        public Type[] GetCallingConventions()
+        {
+            if (_callingConventions == null)
+            {
+                Type[] customModifiers = _returnInfo.GetOptionalCustomModifiers();
+                List<Type>? list = null;
+                bool foundCallingConvention = false;
+
+                for (int i = 0; i < customModifiers.Length; i++)
+                {
+                    Type type = customModifiers[i];
+                    if (type.FullName!.StartsWith(CallingConventionTypePrefix))
+                    {
+                        list ??= new List<Type>();
+                        list.Add(type);
+
+                        if (type == typeof(CallConvCdecl) ||
+                            type == typeof(CallConvFastcall) ||
+                            type == typeof(CallConvStdcall) ||
+                            type == typeof(CallConvThiscall))
+                        {
+                            foundCallingConvention = true;
+                        }
+                    }
+                }
+
+                // Normalize the calling conventions.
+                if (!foundCallingConvention)
+                {
+                    Type? callConv = null;
+
+                    switch (CallingConvention)
+                    {
+                        case MdSigCallingConvention.C:
+                            callConv = typeof(CallConvCdecl);
+                            break;
+                        case MdSigCallingConvention.FastCall:
+                            callConv = typeof(CallConvFastcall);
+                            break;
+                        case MdSigCallingConvention.StdCall:
+                            callConv = typeof(CallConvStdcall);
+                            break;
+                        case MdSigCallingConvention.ThisCall:
+                            callConv = typeof(CallConvThiscall);
+                            break;
+                    }
+
+                    if (callConv != null)
+                    {
+                        list ??= new List<Type>();
+                        list.Add(callConv);
+                        _returnInfo.GetOptionalCustomModifiersList().Add(callConv);
+                    }
+                }
+
+                _callingConventions = list == null ? Type.EmptyTypes : list.ToArray();
+                Debug.Assert(_callingConventions != null);
+            }
+
```

Comment:
How does one tell the difference between managed function pointer and unmanaged function pointer with no calling convention?

Reply:
`Type.IsUnmanagedFunctionPointer`